lib/src/spec/message.dart

```diff
@@ -49,38 +50,71 @@ class MessageData<T> {
   }
 }
 
+/// Delta extension configuration for [MessageExtras]
+class DeltaExtras {
+  /// the id of the message the delta was generated from
+  final String from;
+
+  /// the delta format. Only "vcdiff" is supported currently
+  final String format;
+
+  /// initializes with from and format
+  const DeltaExtras({this.from, this.format});
+
+  /// create instance from a map
+  static DeltaExtras _fromMap(Map value) {
+    if (value == null) return null;
+    return DeltaExtras(
+      from: value['from'] as String,
+      format: value['format'] as String,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeltaExtras && other.from == from && other.format == format;
+
+  @override
+  int get hashCode => '$from:$format'.hashCode;
+}
+
 /// Handles supported message extras types, their encoding and decoding
-class MessageExtras<T> {
-  final T _extras;
+class MessageExtras {
+  /// json-encodable map of extras
+  final Map<String, dynamic> extras;
+
+  /// configuration for delta compression extension
+  DeltaExtras _delta;
 
-  /// Only Map and List types are supported
-  MessageExtras(this._extras) : assert(T == Map);
+  /// delta configuration received from channel message
+  DeltaExtras get delta => _delta;
 
-  /// retrieve extras
-  T get extras => _extras;
+  /// extras is a mandatory json-encodable argument while delta is an extension
+  ///  which defines configuration for delta compression extension
+  MessageExtras(this.extras);
 
   /// initializes [MessageExtras] with given value and validates
   /// the data type, runtime
-  static MessageExtras fromValue(Object value) {
-    if (value == null) {
-      return null;
-    }
-    assert(
-      value is MessageExtras || value is Map,
-      'Message extras must be either `Map`, or `null`.'
-      ' Does not support $value ("${value.runtimeType}")',
-    );
-    if (value is MessageExtras) {
-      return value;
-    } else if (value is Map) {
-      return MessageExtras<Map>(value);
-    } else {
-      throw AssertionError(
-        'Message extras must be either `Map`, or `null`.'
-        ' Does not support $value ("${value.runtimeType}")',
-      );
-    }
+  static MessageExtras fromMap(Map<String, dynamic> value) {
```

Comment:
Actually this is fine, or alternatively use `map` instead of `value`, because that is more readable in the body of the method. I don't mind either way, you can resolve this whenever.

Reply:
renamed here: 16b10b18550155c054f65d2d912ba1051e14ac5c